market/src/contract.rs

```diff
@@ -48,7 +48,16 @@ impl Market {
         let mut count = 0;
 
         for market_option in &self.data.options {
-            let args = Base64VecU8(json!({ "response": count }).to_string().into_bytes());
+            let proposal_id = self.get_random_proposal_id();
+
+            self.proposals.push(proposal_id);
+
+            let args = Base64VecU8(
+                json!({ "proposal_id": proposal_id })
+                    .to_string()
+                    .into_bytes(),
+            );
```

Comment:
@susguzman to keep the players' balance and to be able to `resolve` and `withdraw`, I propose we generate the `proposal_id` as a random string.